src/app/presentation/introjs-directive/introjs.directive.ts

```diff
@@ -19,13 +22,29 @@ export class IntrojsDirective implements OnInit {
       // check if both tours ran TODO: unhardcode this check
       // if (active && localStorage.numTours <= 2) {
       if (active) {
-        setTimeout(() => introJs().start(), 1000);
+        setTimeout(() => {
+          if (this.introJsBefore) {
+            this.introJsBefore();
+          }
+
+          const tour = introJs();
+          tour.start();
+          tour.oncomplete(function () {
+            tour.addHints();
+          });
+        }, 1000);
+
         // localStorage.numTours = +localStorage.numTours + 1;
 
         // @HostListener('keydown') handleKeyboardEvent(eventData: Event) {
         //   alert("eventData");
         // }
+      } else {
+        if (this.introJsAfter) {
+          this.introJsAfter();
```

Comment:
What is the purpose of this else block?

It reads: "If this slide is not active, call introJsAfter". 
When is this branch run at runtime?